app/models/incident.rb

```diff
@@ -7,7 +7,17 @@ class Incident < ApplicationRecord
 
   accepts_nested_attributes_for :incident_competitions, allow_destroy: true
 
-  scope :resolved, -> { where.not(resolved_at: nil) }
+  attribute :visibility, :integer, default: :visible
+
+  enum visibility: {
+    draft: 0,          # Only visible to WRC members
+    staff: 1,          # Visible to WCA staff members only
+    visible: 2,        # Visible to everyone
+  }
+
+  scope :publicly_visible, -> { where(visibility: :visible) }
+  scope :staff_visible, -> { where(visibility: [:staff, :visible]) }
+  scope :wrc_visible, -> { all }
```

Comment:
Rails `enum` gives you scopes by default that you can reuse here. Read more at https://api.rubyonrails.org/v7.1/classes/ActiveRecord/Enum.html